#1593165508 Refactor Authors Haven codebase to ES6

mppdg opened the pull request:
#### What does this PR do?
- This PR converts the existing codebase from ES5 to ES6.

#### Description of Task to be completed?
- install and configure babel to transpile the codebase
- refactor codes in each file to ES6 in the existing codebase
- rebase and refactor new merges to ES6

#### How should this be manually tested?
- `npm start`

#### Any background context you want to provide?
- The refactoring includes existing codebase and PRs that have been merged.  We advise that others write their code in ES6 style going forward.

#### What are the relevant pivotal tracker stories?
- #1593165508

megame24 commented on index.js:
```diff
@@ -70,16 +67,17 @@ if (!isProduction) {
 // production error handler
 // no stacktraces leaked to user
-app.use(function (err, req, res, next) {
+app.use((err, req, res) => {
```
you omitted the `next` parameter in the refactored middleware.

JCanaks replied:
Apart from Resolving the hound errors and including the next parameter as stated in the comment above. I believe the PR is good to go.

mppdg committed:
chore(refactorToES6) Refactor Authors Haven codebase to ES6
- install and configure babel transpiller
- refactor each file to ES6 in the existing codebase
- rebase and refactor new merges to ES6
[Delivers #1593165508]

megame24 commented on routes/api/users.js:
```diff
@@ -1,87 +1,87 @@
-router.post("/users", function(req, res, next) {
-    const username = req.body.user.username,
+router.post('/users', (req, res, next) => {
+  const username = req.body.user.username,
```
You should take care of this destructuring prompt for `const username = req.body.user.username`

rhenshaw56 reviewed (changes requested):
you still have some hound violations, please fix those. there should be no violations on this pr

tutugodfrey commented on index.js:
```diff
@@ -1,90 +1,83 @@
 // will print stacktrace
 if (!isProduction) {
-  app.use(function (err, req, res, next) {
+  app.use((err, req, res, next) => { //eslint-disable-line no-unused-vars
```
Please add the space after forword slashs `//` when commenting so hound won't cry

tutugodfrey commented on config/passport.js:
```diff
@@ -1,26 +1,31 @@
+        if (!user || !validPassword(password)) {
+          return done(null, false, {
+            errors: { 'email or password': 'is invalid' },
+          });
```
Can your try to clear out the indentation errors.

chidioguejiofor reviewed (commented):
Looks good to me once the hound comments  are checked
